tools/src/main/java/org/apache/kafka/tools/BrokerApiVersionsCommand.java

```diff
@@ -0,0 +1,337 @@
+        private static final int DEFAULT_CONNECTION_MAX_IDLE_MS = 9 * 60 * 1000;
+        private static final int DEFAULT_REQUEST_TIMEOUT_MS = 5000;
+        private static final int DEFAULT_MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION = 100;
+        private static final int DEFAULT_RECONNECT_BACKOFF_MS = 50;
+        private static final int DEFAULT_RECONNECT_BACKOFF_MAX = 50;
+        private static final int DEFAULT_SEND_BUFFER_BYTES = 128 * 1024;
+        private static final int DEFAULT_RECEIVE_BUFFER_BYTES = 32 * 1024;
+        private static final int DEFAULT_RETRY_BACKOFF_MS = 100;
+
+        private static final AtomicInteger ADMIN_CLIENT_ID_SEQUENCE = new AtomicInteger(1);
+        private static final ConfigDef ADMIN_CONFIG_DEF = new ConfigDef()
+                .define(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, ConfigDef.Type.LIST, ConfigDef.Importance.HIGH, CommonClientConfigs.BOOTSTRAP_SERVERS_DOC)
+                .define(CommonClientConfigs.CLIENT_DNS_LOOKUP_CONFIG, ConfigDef.Type.STRING, ClientDnsLookup.USE_ALL_DNS_IPS.toString(), ConfigDef.ValidString.in(ClientDnsLookup.USE_ALL_DNS_IPS.toString(), ClientDnsLookup.RESOLVE_CANONICAL_BOOTSTRAP_SERVERS_ONLY.toString()), ConfigDef.Importance.MEDIUM, CommonClientConfigs.CLIENT_DNS_LOOKUP_DOC)
+                .define(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, ConfigDef.Type.STRING, CommonClientConfigs.DEFAULT_SECURITY_PROTOCOL, ConfigDef.CaseInsensitiveValidString.in(Utils.enumOptions(SecurityProtocol.class)), ConfigDef.Importance.MEDIUM, CommonClientConfigs.SECURITY_PROTOCOL_DOC)
+                .define(CommonClientConfigs.REQUEST_TIMEOUT_MS_CONFIG, ConfigDef.Type.INT, DEFAULT_REQUEST_TIMEOUT_MS, ConfigDef.Importance.MEDIUM, CommonClientConfigs.REQUEST_TIMEOUT_MS_DOC)
+                .define(CommonClientConfigs.SOCKET_CONNECTION_SETUP_TIMEOUT_MS_CONFIG, ConfigDef.Type.LONG, CommonClientConfigs.DEFAULT_SOCKET_CONNECTION_SETUP_TIMEOUT_MS, ConfigDef.Importance.MEDIUM, CommonClientConfigs.SOCKET_CONNECTION_SETUP_TIMEOUT_MS_DOC)
+                .define(CommonClientConfigs.SOCKET_CONNECTION_SETUP_TIMEOUT_MAX_MS_CONFIG, ConfigDef.Type.LONG, CommonClientConfigs.DEFAULT_SOCKET_CONNECTION_SETUP_TIMEOUT_MAX_MS, ConfigDef.Importance.MEDIUM, CommonClientConfigs.SOCKET_CONNECTION_SETUP_TIMEOUT_MAX_MS_DOC)
+                .define(CommonClientConfigs.RETRY_BACKOFF_MS_CONFIG, ConfigDef.Type.LONG, DEFAULT_RETRY_BACKOFF_MS, ConfigDef.Importance.MEDIUM, CommonClientConfigs.RETRY_BACKOFF_MS_DOC)
+                .withClientSslSupport()
+                .withClientSaslSupport();
+
+        private volatile boolean running = true;
+        private final Time time;
+        private final ConsumerNetworkClient client;
+        private final List<Node> bootstrapBrokers;
+        private final ConcurrentLinkedQueue<RequestFuture<ClientResponse>> pendingFutures = new ConcurrentLinkedQueue<>();
+
+        static class AdminConfig extends AbstractConfig {
```

Comment:
Could you please remove it as we can use `AbstractConfig` directly? for instance:
```java
        static AdminClient create(Map<String, ?> props) {
            return create(new AbstractConfig(ADMIN_CONFIG_DEF, props, false));
        }
```